### ISSUE TITLE
Displaying additional information in 'Add Entry' page?

### ISSUE THREAD

**Issue by embeh** [meeting-room-booking-system/mrbs-code#3508]
For certain bookable items, we would like to display special instructions, warning etc. in the 'Add Entry' page, so that our users are informed about any special things to consider while creating a new booking entry.

--> What would be the appropriate place/mechanism to do add such text (ideally with a bit of HTML styling)?

We are currently running MRBS 1.8.0 but any pointer for newer release might also help me getting started.

Thank you!

**Comment by campbell-m** [comment]
In MRBS 1.8.0 edit the file edit_entry.php and at line 1733 add

~~~php
// Add a note (which can contain HTML and so is not escaped)
$note_token = 'edit_entry_note';
$note_text = get_vocab($note_token);
if (($note_text !== '') && ($note_text != $note_token))
{
  $note = new ElementDiv();
  $note->setAttribute('id', 'edit_entry_note')
       ->setText($note_text, true, true);
  $fieldset->addElement($note);
}
~~~

and then in your config file set

~~~php
$vocab_override["en"]["edit_entry_note"]  = "<span>Special instructions</span>"; // For example
~~~

and you can style it with CSS.  It will be inside a div with id "edit_entry_note".

I might look at adding this to MRBS as it would be a fairly common requirement.

**Comment by embeh** [comment]
Thanks very much for the speedy reply!
From the code I cannot see immediately how this would allow to show a different note for different items and/or different areas?

Adding this to MRBS at large seems very useful to me, as often an item needs additional instructions (how to use it, e.g. with a web link to full instructions, or when to be careful, etc...)

**Comment by campbell-m** [comment]
Ah, OK, I'd missed the bit that it was room-specific.   That makes it a bit more complicated, because although it would be fairly easy to get the instructions from the room table in the database, you then have to think about what to do when the user selects a different room from the room select drop-down: you'd need to change the note - either by having all the notes pre-loaded in advance, or else by getting the appropriate note in an Ajax request.

Yes, you could create a feature request for this, though I'm not sure when it would be implemented.

**Comment by embeh** [comment]
I see! 

I actually never realized that the 'Add Entry' page has a menu to select which item to book! 
I always ignored that because I always go to the week (or month) view, then click on an empty spot and the menu is preselected with the right item. In three years of usage, I never changed that preselected item...

So I expected the 'Add Entry' form to already be item-specific...

**Comment by campbell-m** [comment]
Yes, it's possible though.  We already have to do something similar when the area is changed.  In that case not only does the list of rooms have to change, but so do the possible timeslots.

**Comment by OscarERodriguezSaldivar** [comment]
been stuck on this since yesterday and jumped straight to modify the code without checking the documentation, thank you!! and sorry

**Comment by ProjectHomeUsingGit** [comment]
Good day: Thanks @campbell-m for always resolving and responding to our issues in a timely manner. In regard to your answer to the question, I have version 1.10 installed and wasn't sure if the line number you are referring to for 1.8 would be the same for 1.10. Just wanted to confirm. Thank you.

**Comment by campbell-m** [comment]
In MRBS 1.10.0 the code should be inserted at line 1743, ie just after

~~~php
$fieldset = new ElementFieldset();
$fieldset->addLegend(get_vocab($token));
~~~

By the way, I recommend upgrading to the latest release of MRBS if possible.

**Comment by embeh** [comment]
Related to #3874
